measures/ApplyUpgrade/measure.rb

```diff
@@ -271,14 +272,19 @@ def run(model, runner, user_arguments)
         end
       end
 
+      # Check the size of the measures hash at this point, and halt the workflow if it's empty
       if halt_workflow(runner, measures)
         return false
       end
 
-      if !measures.keys.include?('ResStockArguments') # upgrade is via another measure
+      # Check if upgrade is via another measure
+      if !measures.keys.include?('ResStockArguments')
         measures['ResStockArguments'] = [{}]
       end
 
+      # Save the hash of applicable upgrade measure arguments
+      upgrade_args_hash = measures['ResStockArguments'][0].clone
```

Comment:
Using only _applicable_ arguments (i.e., only those arguments corresponding to options that meet the apply logic) fixes a bug related to:
* retaining HVAC system sizes; previously, it was possible that an HVAC system was being resized if (1) an HVAC option didn't meet apply logic (i.e., was being retained), (2) the upgrade system type was the same as the baseline system type (e.g., ASHP to ASHP), and (3) the upgrade included changes in design loads (e.g., envelope).
* identifying panel new loads; previously, existing loads were being set as new loads (e.g., electric clothes dryer in the baseline building)

Reply:
Am I interpreting correctly that the bug also affected the load-based calculation?

Reply:
Yes.